mlir/include/mlir/Dialect/LLVMIR/LLVMAttrDefs.td

```diff
@@ -625,10 +625,10 @@ def LLVM_DINamespaceAttr : LLVM_Attr<"DINamespace", "di_namespace",
 def LLVM_DISubrangeAttr : LLVM_Attr<"DISubrange", "di_subrange", /*traits=*/[],
                                     "DINodeAttr"> {
   let parameters = (ins
-    OptionalParameter<"IntegerAttr">:$count,
-    OptionalParameter<"IntegerAttr">:$lowerBound,
-    OptionalParameter<"IntegerAttr">:$upperBound,
-    OptionalParameter<"IntegerAttr">:$stride
+    OptionalParameter<"::mlir::Attribute">:$count,
```

Comment:
I do not recall this, but I guess attribute parameters cannot be constraint through table gen, or can they? Ideally, we would explicitly list which types are legal here.

Reply:
I could not find a simple way to add that constraint so I am open to suggestions here. My initial approach was to add a `DISubrangeValueAttr` type which has fields for all the types that we expect (e.g. `IntegerAttr`, `DIExpressionAttr`, etc). Then use that in the `DISubrangeAttr` in place of `IntegerAttr`. But it seems a bit overkill to me. Also made the usage a bit more cumbersome.

Reply:
The LLVM BoundType is a pointer union. That could be an option that maybe less cumbersome than a separate attribute. Worst case we can also go for a plain attribute.